providers/src/airflow/providers/http/sensors/http.py

```diff
@@ -17,10 +17,14 @@
 # under the License.
 from __future__ import annotations
 
+import base64
+import pickle
```

Comment:
I'm not sure here 🤔  I thought Airflow wanted to avoid pickle as much as possible. @eladkal WDYT?

Reply:
I'll continue reviewing once we agree on whether we want to use pickle 🙂

Reply:
My default answer to "do we want to use pickle?" is "hell no".

This is more nuanced when it comes to details, but using pickle is a HUGE security floodgate and using it for anything but internal passing of data for multiprocessing where it is not visible or even accesible to the external users is "no go" by default. 

Int this case - when we want to pass pickled data as a TriggerEvent - in order for it to be serialized and then unpickled, the answer is "ABSOLUTELY NOT".

Cloudpickle might be a way to go.

Reply:
@Lee-W @potiuk I will change it to `cloudpickle`. 

But I see the `HttpOperator` uses the pickle currently ([link](https://github.com/apache/airflow/blob/9bd1c4064035a4ee206b07055b3daff2700e2e74/providers/src/airflow/providers/http/operators/http.py#L21)) for deferrable. 

Please let me know whether we need to change it

Reply:
I would probably say yes, but not necessary need to be done in this PR

Reply:
changed the pickle library to cloudpickle